logstreams/src/test/java/io/camunda/zeebe/logstreams/impl/log/LogStreamBatchReaderTest.java

```diff
@@ -0,0 +1,335 @@
+
+    // when
+    final var found = batchReader.seekToNextBatch(-1L);
+
+    // then
+    assertThat(found).isTrue();
+    assertThat(batchReader.hasNext()).isTrue();
+
+    final var batch = batchReader.next();
+    assertThat(batch.hasNext()).isTrue();
+    assertThat(batch.next().getPosition()).isEqualTo(eventPosition1);
+  }
+
+  @Test
+  public void shouldSeekToNextBatch() {
+    // given
+    writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    final long eventPosition2 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    final long eventPosition3 = writerRule.sourceEventPosition(2L).writeEvent(EVENT_VALUE);
+
+    // when
+    final var found = batchReader.seekToNextBatch(eventPosition2);
+
+    // then
+    assertThat(found).isTrue();
+    assertThat(batchReader.hasNext()).isTrue();
+
+    final var batch = batchReader.next();
+    assertThat(batch.hasNext()).isTrue();
+    assertThat(batch.next().getPosition()).isEqualTo(eventPosition3);
+  }
+
+  @Test
+  public void shouldSeekToNextEventWithinBatch() {
+    // given
+    final long eventPosition1 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    final long eventPosition2 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    writerRule.sourceEventPosition(2L).writeEvent(EVENT_VALUE);
+
+    // when
+    final var found = batchReader.seekToNextBatch(eventPosition1);
+
+    // then
+    assertThat(found).isTrue();
+    assertThat(batchReader.hasNext()).isTrue();
+
+    final var batch = batchReader.next();
+    assertThat(batch.hasNext()).isTrue();
+    assertThat(batch.next().getPosition()).isEqualTo(eventPosition2);
+  }
+
+  @Test
+  public void shouldSeekToTailIfLastEvent() {
+    // given
+    writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+    final long eventPosition2 = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+
+    // when
+    final var found = batchReader.seekToNextBatch(eventPosition2);
+
+    // then
+    assertThat(found).isTrue();
+    assertThat(batchReader.hasNext()).isFalse();
+  }
+
+  @Test
+  public void shouldSeekToNotExistingPosition() {
+    // given
+    final var eventPosition = writerRule.sourceEventPosition(1L).writeEvent(EVENT_VALUE);
+
+    // when
+    final var found = batchReader.seekToNextBatch(eventPosition + 1);
+
+    // then
+    assertThat(found).isFalse();
+    assertThat(batchReader.hasNext()).isFalse();
+  }
+
+  @Test
+  public void shouldNotHaveNextIfClosed() {
+    // given
+    batchReader.close();
+
+    // when - then
+    assertThat(batchReader.hasNext()).isFalse();
+  }
+
+  @Test
+  public void shouldThrowNoSuchElementExceptionOnNextBatch() {
+    // given
+    assertThat(batchReader.hasNext()).isFalse();
+
+    // when / then
+    assertThatThrownBy(batchReader::next).isInstanceOf(NoSuchElementException.class);
+  }
+
+  @Test
+  public void shouldThrowNoSuchElementExceptionOnNextEvent() {
```

Comment:
Batch should also thrown no such element if next is called right?

Reply:
Correct. Do you expect something else?

Reply:
I expect an test then